src/OptionSettlement.sol

```diff
@@ -188,31 +188,52 @@ contract OptionSettlementEngine is ERC1155, IOptionSettlementEngine {
 
     /// @inheritdoc IOptionSettlementEngine
     function write(uint256 optionId, uint112 amount) external returns (uint256 claimId) {
-        if (amount == 0) {
-            revert AmountWrittenCannotBeZero();
-        }
+        /// claimId == 0 signifies that a new claim NFT should be minted
```

Comment:
Where is this claimid coming from? Will it ever be 0?

Reply:
A new claim should always be minted if one is not provided.

Reply:
this is what's happening here, I think the comment is just misleading